os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -437,6 +509,25 @@ trwifi_result_e wifi_netmgr_utils_connect_ap(struct netdev *dev, trwifi_ap_confi
 		}
 	}
 
+	ap_channel = 0xffff;
+
+	if (scan_number) {
+		int i;
+		rtw_cancel_timer(&(scan_timer));

```

Comment:
Why does it cancel timer when it try to connect???

Reply:
It is to make sure that the timer is not expired during the ap connection.
if the timer is running, it may have possibility of timer expired.

Reply:
> 		if (scan_number)
			rtw_set_timer(&(scan_timer), SCAN_TIMER_DURATION);
Then it is possible to extend lifetime of scan_list forever with connect try.
How about using mutex for shared resource (saved_scan_list)??

Reply:
@j11591 , The rtw_enter_critical/rtw_exit_critical API's can be added to prevent scan timer timeout(which frees up  saved_scan_list and delete the timer). Are there any more review comments? I'll update all in single commit.

Reply:
@keerthanalex Before canceling the timer, I think you need to store the remaining duration and to reset the timer with this duration after connecting AP.

Reply:
The rtw_enter_critical/rtw_exit_critical API's may be excessive, so we plan to go with mutex to protect the `saved_scan_list`.
@davidfather, we only need to protect the list from getting deleted by timeout. Hence we can skip stop/reset timer and use mutex for this case.
But we would to like to get suggestion on adding `rtw_mutext_init` where it needs to be called only once before using `rtw_mutext_get()` and `rtw_mutext_put()`.